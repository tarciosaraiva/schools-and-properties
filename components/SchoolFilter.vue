<template>
  <modal @close="closeFn">
    <template #header>
      <h2>School filter</h2>
    </template>
    <template #body>
  <form>
      <div class="pl-row">
        <div class="pl-col">
          <label for="s_edu_sector">Education sector</label>
          <select id="s_edu_sector" v-model="sEducationSector">
            <option value="all">All</option>
            <option value="Government">Government</option>
            <option value="NonGovernment">Non Government</option>
          </select>
        </div>
      </div>
      <div class="pl-row">
        <div class="pl-col">
          <label for="s_school_zones">School zones</label>
          <select id="s_school_zones" v-model="sZone">
            <option value="none">None</option>
            <option value="primary">Primary</option>
            <option value="p7">Secondary Yr 7</option>
            <option value="p8">Secondary Yr 8</option>
            <option value="p9">Secondary Yr 9</option>
            <option value="p10">Secondary Yr 10</option>
            <option value="p11">Secondary Yr 11</option>
            <option value="p12">Secondary Yr 12</option>
          </select>
        </div>
      </div>
      <div class="pl-row">
        <div class="pl-col">
          <label for="s_show_primary_schools" class="checkbox">
          <input
            id="s_show_primary_schools"
            v-model="sPlotPrimary"
            type="checkbox"
          /> Show primary schools</label>
        </div>
      </div>
      <div class="pl-row">
        <div class="pl-col">
          <label for="s_prim_rating">Overall score</label>
          <select id="s_prim_rating" v-model="sPrimRating">
            <option value="100">= 100</option>
            <option value="99">>= 99</option>
            <option value="98">>= 98</option>
            <option value="97">>= 97</option>
            <option value="96">>= 96</option>
            <option value="95">>= 95</option>
            <option value="94">>= 94</option>
            <option value="93">>= 93</option>
            <option value="92">>= 92</option>
            <option value="91">>= 91</option>
            <option value="90">>= 90</option>
          </select>
        </div>
      </div>
      <div class="pl-row">
        <div class="pl-col">
          <label for="s_prim_english_rating">English score</label>
          <select id="s_prim_english_rating" v-model="sPrimEnglishRating">
            <option value="5">= 5</option>
            <option value="4">>= 4</option>
            <option value="3">>= 3</option>
            <option value="2">>= 2</option>
            <option value="1">>= 1</option>
          </select>
        </div>
        <div class="pl-col">
          <label for="s_prim_maths_rating">Maths score</label>
          <select id="s_prim_maths_rating" v-model="sPrimMathsRating">
            <option value="5">= 5</option>
            <option value="4">>= 4</option>
            <option value="3">>= 3</option>
            <option value="2">>= 2</option>
            <option value="1">>= 1</option>
          </select>
        </div>
      </div>
      <div class="pl-row">
        <div class="pl-col">
          <label for="s_show_secondary_schools">
          <input
            id="s_show_secondary_schools"
            v-model="sPlotSecondary"
            type="checkbox" /> Show secondary schools</label>
        </div>
      </div>
      <div class="pl-row">
        <div class="pl-col">
          <label for="s_sec_rating">Overall score</label>
          <select id="s_sec_rating" v-model="sSecRating">
            <option value="100">= 100</option>
            <option value="99">>= 99</option>
            <option value="98">>= 98</option>
            <option value="97">>= 97</option>
            <option value="96">>= 96</option>
            <option value="95">>= 95</option>
            <option value="94">>= 94</option>
            <option value="93">>= 93</option>
            <option value="92">>= 92</option>
            <option value="91">>= 91</option>
            <option value="90">>= 90</option>
          </select>
        </div>
      </div>
      <div class="pl-row">
        <div class="pl-col">
          <label for="s_sec_english_rating">English score</label>
          <select id="s_sec_english_rating" v-model="sSecEnglishRating">
            <option value="5">= 5</option>
            <option value="4">>= 4</option>
            <option value="3">>= 3</option>
            <option value="2">>= 2</option>
            <option value="1">>= 1</option>
          </select>
        </div>
        <div class="pl-col">
          <label for="s_sec_maths_rating">Maths score</label>
          <select id="s_sec_maths_rating" v-model="sSecMathsRating">
            <option value="5">= 5</option>
            <option value="4">>= 4</option>
            <option value="3">>= 3</option>
            <option value="2">>= 2</option>
            <option value="1">>= 1</option>
          </select>
        </div>
      </div>
  </form>
    </template>
  </modal>
</template>

<script lang="ts">
import Vue from 'vue'
import { mapGetters } from 'vuex'

import Modal from '~/components/Modal.vue'

export default Vue.extend({
  components: { Modal },
  props: {
    closeFn: Function
  },
  computed: {
    ...mapGetters(['schoolsFilter']),
    sEducationSector: {
      get() {
        return this.schoolsFilter.educationSector
      },
      set(value) {
        this.$store.commit('UPDATE_SCHOOL_EDUCATION_SECTOR', value)
      },
    },
    sZone: {
      get() {
        return this.schoolsFilter.zone
      },
      set(value) {
        this.$store.commit('UPDATE_SCHOOL_ZONE', value)
      },
    },
    sPlotPrimary: {
      get() {
        return this.schoolsFilter.primary.plot
      },
      set(value) {
        this.$store.commit('UPDATE_PLOT_PRIMARY_SCHOOLS', value)
      },
    },
    sPrimRating: {
      get() {
        return this.schoolsFilter.primary.rating
      },
      set(value) {
        this.$store.commit('UPDATE_SCHOOL_PRIMARY_RATING', value)
      },
    },
    sPrimEnglishRating: {
      get() {
        return this.schoolsFilter.primary.englishRating
      },
      set(value) {
        this.$store.commit('UPDATE_SCHOOL_PRIMARY_ENG_RATING', value)
      },
    },
    sPrimMathsRating: {
      get() {
        return this.schoolsFilter.primary.mathsRating
      },
      set(value) {
        this.$store.commit('UPDATE_SCHOOL_PRIMARY_MATH_RATING', value)
      },
    },
    sPlotSecondary: {
      get() {
        return this.schoolsFilter.secondary.plot
      },
      set(value) {
        this.$store.commit('UPDATE_PLOT_SECONDARY_SCHOOLS', value)
      },
    },
    sSecRating: {
      get() {
        return this.schoolsFilter.secondary.rating
      },
      set(value) {
        this.$store.commit('UPDATE_SCHOOL_SECONDARY_RATING', value)
      },
    },
    sSecEnglishRating: {
      get() {
        return this.schoolsFilter.secondary.englishRating
      },
      set(value) {
        this.$store.commit('UPDATE_SCHOOL_SECONDARY_ENG_RATING', value)
      },
    },
    sSecMathsRating: {
      get() {
        return this.schoolsFilter.secondary.mathsRating
      },
      set(value) {
        this.$store.commit('UPDATE_SCHOOL_SECONDARY_MATH_RATING', value)
      },
    },
  }
})
</script>
