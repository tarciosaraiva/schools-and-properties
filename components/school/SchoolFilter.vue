<template>
  <modal @close="closeFn">
    <template #header>
      <h2>School filter</h2>
    </template>
    <template #body>
  <form>
      <div class="pl-row">
        <div class="pl-col">
          <label for="s_edu_sector">Education sector</label>
          <select id="s_edu_sector" v-model="sEducationSector">
            <option value="all">All</option>
            <option value="Government">Government</option>
            <option value="NonGovernment">Non Government</option>
          </select>
        </div>
      </div>
      <div class="pl-row">
        <div class="pl-col">
          <label for="s_show_primary_schools" class="checkbox">
          <input
            id="s_show_primary_schools"
            v-model="sPlotPrimary"
            type="checkbox"
          /> Show primary schools</label>
        </div>
      </div>
      <div class="pl-row">
        <div class="pl-col">
          <label for="s_prim_rating">Overall score</label>
          <select id="s_prim_rating" v-model="sPrimRating">
            <option value="100">= 100</option>
            <option value="99">>= 99</option>
            <option value="98">>= 98</option>
            <option value="97">>= 97</option>
            <option value="96">>= 96</option>
            <option value="95">>= 95</option>
            <option value="94">>= 94</option>
            <option value="93">>= 93</option>
            <option value="92">>= 92</option>
            <option value="91">>= 91</option>
            <option value="90">>= 90</option>
            <option value="0">All schools</option>
          </select>
        </div>
      </div>
      <div class="pl-row">
        <div class="pl-col">
          <label for="s_show_secondary_schools">
          <input
            id="s_show_secondary_schools"
            v-model="sPlotSecondary"
            type="checkbox" /> Show secondary schools</label>
        </div>
      </div>
      <div class="pl-row">
        <div class="pl-col">
          <label for="s_sec_rating">Overall score</label>
          <select id="s_sec_rating" v-model="sSecRating">
            <option value="100">= 100</option>
            <option value="99">>= 99</option>
            <option value="98">>= 98</option>
            <option value="97">>= 97</option>
            <option value="96">>= 96</option>
            <option value="95">>= 95</option>
            <option value="94">>= 94</option>
            <option value="93">>= 93</option>
            <option value="92">>= 92</option>
            <option value="91">>= 91</option>
            <option value="90">>= 90</option>
            <option value="0">All schools</option>
          </select>
        </div>
      </div>
  </form>
    </template>
  </modal>
</template>

<script lang="ts">
import Vue from 'vue'
import { mapGetters } from 'vuex'

import Modal from '~/components/Modal.vue'

export default Vue.extend({
  components: { Modal },
  props: {
    closeFn: Function
  },
  computed: {
    ...mapGetters(['schoolsFilter']),
    sEducationSector: {
      get() {
        return this.schoolsFilter.educationSector
      },
      set(value) {
        this.$store.commit('UPDATE_SCHOOL_EDUCATION_SECTOR', value)
      },
    },
    sPlotPrimary: {
      get() {
        return this.schoolsFilter.primary.plot
      },
      set(value) {
        this.$store.commit('UPDATE_PLOT_PRIMARY_SCHOOLS', value)
      },
    },
    sPrimRating: {
      get() {
        return this.schoolsFilter.primary.rating
      },
      set(value) {
        this.$store.commit('UPDATE_SCHOOL_PRIMARY_RATING', value)
      },
    },
    sPlotSecondary: {
      get() {
        return this.schoolsFilter.secondary.plot
      },
      set(value) {
        this.$store.commit('UPDATE_PLOT_SECONDARY_SCHOOLS', value)
      },
    },
    sSecRating: {
      get() {
        return this.schoolsFilter.secondary.rating
      },
      set(value) {
        this.$store.commit('UPDATE_SCHOOL_SECONDARY_RATING', value)
      },
    },
  }
})
</script>
