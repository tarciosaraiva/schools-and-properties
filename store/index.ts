import { GetterTree, ActionTree, MutationTree } from 'vuex'

interface ApplicationFilter {
  properties: PropertiesFilter,
  schools: SchoolsFilter
}

interface PropertiesFilter {
  types: string[],
  rooms: number,
  bathrooms: number,
  carSpaces: number,
  maxPrice: number
}

export interface SchoolsFilter {
  educationSector: string,
  zone: string,
  primary: SchoolLevelRating,
  secondary: SchoolLevelRating,
}

interface SchoolLevelRating {
  plot: boolean,
  rating: number,
  englishRating: number,
  mathsRating: number
}

export interface PropertyListing {
  id: number,
  price: string,
  area: string
  bathrooms: number,
  bedrooms: number,
  carspaces: number,
  displayableAddress: string,
  features: string[],
  landArea: number,
  latitude: number,
  longitude: number,
  postcode: string,
  propertyType: string
  region: string,
  state: string,
  street: string,
  streetNumber: string,
  suburb: string,
  unitNumber: string,
  media: string[],
  listingSlug: string
}

export interface School {
  schoolName: string,
  educationSector: string,
  schoolType: string,
  phoneNumber: string,
  primaryOverallScore: string,
  primaryEnglishScore: string,
  primaryMathsScore: string,
  primaryEnrolments: string,
  secondaryOverallScore: string,
  secondaryEnglishScore: string,
  secondaryMathsScore: string,
  secondaryEnrolments: string
}

export const state = () => ({
  bounds: {
    ne: { lat: 0, lon: 0 },
    nw: { lat: 0, lon: 0 },
    se: { lat: 0, lon: 0 },
    sw: { lat: 0, lon: 0 },
  },
  geocodeSuggestions: [] as any,
  list: [] as PropertyListing[],
  filter: {
    properties: {
      types: ['House', 'Townhouse'],
      rooms: 3,
      bathrooms: 2,
      carSpaces: 1,
      maxPrice: 900000
    },
    schools: {
      educationSector: 'all',
      zone: 'none',
      primary: {
        plot: true,
        rating: 95,
        englishRating: 4,
        mathsRating: 4
      },
      secondary: {
        plot: true,
        rating: 95,
        englishRating: 4,
        mathsRating: 4
      }
    }
  } as ApplicationFilter
})

export type RootState = ReturnType<typeof state>

export const getters: GetterTree<RootState, RootState> = {
  listings: state => state.list,
  schoolsFilter: state => state.filter.schools,
  propertiesFilter: state => state.filter.properties,
  locations: state => state.geocodeSuggestions,
  listingPois: state => state.list.map(l => ({
    type: 'Feature',
    id: l.id,
    geometry: {
      type: 'Point',
      coordinates: [l.longitude, l.latitude]
    },
    properties: { ...l }
  }))
}

export const mutations: MutationTree<RootState> = {
  ADD_LISTINGS (state, listings: any[]) {
    const propertyListings = listings
      .filter(l => l.type === 'PropertyListing')
      .map(({ listing }) => {
        const propertyListing: PropertyListing = {
          id: listing.id,
          price: listing.priceDetails?.displayPrice,
          area: listing.propertyDetails.area,
          bathrooms: listing.propertyDetails.bathrooms,
          bedrooms: listing.propertyDetails.bedrooms,
          carspaces: listing.propertyDetails.carspaces,
          displayableAddress: listing.propertyDetails.displayableAddress,
          features: listing.propertyDetails.features,
          landArea: listing.propertyDetails.landArea,
          latitude: listing.propertyDetails.latitude,
          longitude: listing.propertyDetails.longitude,
          postcode: listing.propertyDetails.postcode,
          propertyType: listing.propertyDetails.propertyType,
          region: listing.propertyDetails.region,
          state: listing.propertyDetails.state,
          street: listing.propertyDetails.street,
          streetNumber: listing.propertyDetails.streetNumber,
          suburb: listing.propertyDetails.suburb,
          unitNumber: listing.propertyDetails.unitNumber,
          media: listing.media.map((m: any) => m.url),
          listingSlug: listing.listingSlug
        }

        return propertyListing
      })

    state.list = propertyListings
  },
  UPDATE_PROPERTY_FILTER_TYPE (state, payload) {
    state.filter.properties.types = payload
  },
  UPDATE_PROPERTY_FILTER_ROOMS (state, payload) {
    state.filter.properties.rooms = Number(payload)
  },
  UPDATE_PROPERTY_FILTER_BATHROOMS (state, payload) {
    state.filter.properties.bathrooms = Number(payload)
  },
  UPDATE_PROPERTY_FILTER_CAR_SPACES (state, payload) {
    state.filter.properties.carSpaces = Number(payload)
  },
  UPDATE_PROPERTY_FILTER_MAX_PRICE (state, payload) {
    state.filter.properties.maxPrice = Number(payload)
  },
  UPDATE_BOUNDING_BOX (state, payload) {
    state.bounds.nw.lat = payload.getNorthWest().lat
    state.bounds.nw.lon = payload.getNorthWest().lng
    state.bounds.ne.lat = payload.getNorthEast().lat
    state.bounds.ne.lon = payload.getNorthEast().lng
    state.bounds.se.lat = payload.getSouthEast().lat
    state.bounds.se.lon = payload.getSouthEast().lng
    state.bounds.sw.lat = payload.getSouthWest().lat
    state.bounds.sw.lon = payload.getSouthWest().lng
  },
  UPDATE_SCHOOL_EDUCATION_SECTOR (state, payload) {
    state.filter.schools.educationSector = payload
  },
  UPDATE_SCHOOL_ZONE (state, payload) {
    state.filter.schools.zone = payload
  },
  UPDATE_PLOT_PRIMARY_SCHOOLS (state, payload) {
    state.filter.schools.primary.plot = payload
  },
  UPDATE_SCHOOL_PRIMARY_RATING (state, payload) {
    state.filter.schools.primary.rating = Number(payload)
  },
  UPDATE_SCHOOL_PRIMARY_ENG_RATING (state, payload) {
    state.filter.schools.primary.englishRating = Number(payload)
  },
  UPDATE_SCHOOL_PRIMARY_MATH_RATING (state, payload) {
    state.filter.schools.primary.mathsRating = Number(payload)
  },
  UPDATE_PLOT_SECONDARY_SCHOOLS (state, payload) {
    state.filter.schools.secondary.plot = payload
  },
  UPDATE_SCHOOL_SECONDARY_RATING (state, payload) {
    state.filter.schools.secondary.rating = Number(payload)
  },
  UPDATE_SCHOOL_SECONDARY_ENG_RATING (state, payload) {
    state.filter.schools.secondary.englishRating = Number(payload)
  },
  UPDATE_SCHOOL_SECONDARY_MATH_RATING (state, payload) {
    state.filter.schools.secondary.mathsRating = Number(payload)
  },
  ADD_GEOCODE_RESULTS (state, payload) {
    state.geocodeSuggestions = payload.map((f: any) => ({
      center: f.center,
      name: f.place_name,
      placeType: f.place_type
    }))
  }
}

export const actions: ActionTree<RootState, RootState> = {
  async loadListings ({ commit, state }) {
    const { filter, bounds } = state

    const payload = {
      listingType: 'Sale',
      pageSize: 200,
      propertyTypes: filter.properties.types,
      minBedrooms: filter.properties.rooms,
      minBathrooms: filter.properties.bathrooms,
      minCarspaces: filter.properties.carSpaces,
      maxPrice: filter.properties.maxPrice,
      locations: [{ state: 'VIC' }],
      geoWindow: { box: { topLeft: bounds.nw, bottomRight: bounds.se } }
    }

    const listings = await this.$axios.$post('https://api.domain.com.au/v1/listings/residential/_search', payload, { headers: {'X-Api-Key': process.env.domainApiKey } })
    commit('ADD_LISTINGS', listings)
  },

  async geocode ({ commit }, searchTerm) {
    if (searchTerm) {
      const params = {
        key: process.env.mapTilerSecret,
        bbox: '141.24847451171752,-39.624732230379166,148.68072548828115,-34.32302549095252',
        language: 'en'
      }

      const geocodeResults = await this.$axios.$get(`https://api.maptiler.com/geocoding/${searchTerm}.json`, { params })
      const schoolsResults = await this.$axios.$get(`//${process.env.domain}/schools/schools.json`)
      const mappedSchoolResults = schoolsResults.map((s: any) => ({
        center: [s.lon, s.lat],
        place_name: s.schoolName,
        place_type: 'school'
      })).filter((s: any) => s.place_name.toLowerCase().includes(searchTerm))

      commit('ADD_GEOCODE_RESULTS', [...geocodeResults.features, ...mappedSchoolResults])
    } else {
      commit('ADD_GEOCODE_RESULTS', [])
    }
  },

  setBoundingBox({ commit }, payload) {
    commit('UPDATE_BOUNDING_BOX', payload)
  }
}
