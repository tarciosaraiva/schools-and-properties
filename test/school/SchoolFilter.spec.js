import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'

import { state, getters } from '@/store'
import SchoolFilter from '~/components/school/SchoolFilter.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('SchoolFilter', () => {
  let store
  let wrapper
  let propsData
  let mutations

  beforeEach(() => {
    propsData = { closeFn: jest.fn() }
    mutations = {
      'UPDATE_PLOT_PRIMARY_SCHOOLS': jest.fn(),
      'UPDATE_PLOT_SECONDARY_SCHOOLS': jest.fn(),
      'UPDATE_SCHOOL_EDUCATION_SECTOR': jest.fn(),
      'UPDATE_SCHOOL_PRIMARY_RATING': jest.fn(),
      'UPDATE_SCHOOL_SECONDARY_RATING': jest.fn(),
    }
    store = new Vuex.Store({ state, mutations, getters })
    wrapper = mount(SchoolFilter, { propsData, store, localVue })
  })

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly with props and store', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('education sector default value matches state', () => {
    const input = wrapper.find('#s_edu_sector')
    expect(input.element.value).toBe(store.state.filter.schools.educationSector)
  })

  test('overall rating default value match state', () => {
    const input = wrapper.find('#s_prim_rating')
    expect(Number(input.element.value)).toBe(store.state.filter.schools.primary.rating)
  })

  test('should have committed mutation "UPDATE_PLOT_PRIMARY_SCHOOLS" to the store when plotting option changes', async () => {
    const input = wrapper.find('#s_show_primary_schools')
    await input.setChecked(false)
    expect(mutations.UPDATE_PLOT_PRIMARY_SCHOOLS).toHaveBeenCalledWith(store.state, false)
  })

  test('should have committed mutation "UPDATE_PLOT_SECONDARY_SCHOOLS" to the store when plotting option changes', async () => {
    const input = wrapper.find('#s_show_secondary_schools')
    await input.setChecked(false)
    expect(mutations.UPDATE_PLOT_SECONDARY_SCHOOLS).toHaveBeenCalledWith(store.state, false)
  })

  test.each([
    ['UPDATE_SCHOOL_EDUCATION_SECTOR', '#s_edu_sector', 'NonGovernment'],
    ['UPDATE_SCHOOL_PRIMARY_RATING', '#s_prim_rating', '99'],
    ['UPDATE_SCHOOL_SECONDARY_RATING', '#s_sec_rating', '98'],
  ])(
    'should have committed mutation %p to the store when rating changes',
    (mutation, elementId, value) => {
      const input = wrapper.find(elementId)
      input.element.value = value
      input.trigger('change')

      expect(mutations[mutation]).toHaveBeenCalledWith(store.state, value)
    }
  )
})
