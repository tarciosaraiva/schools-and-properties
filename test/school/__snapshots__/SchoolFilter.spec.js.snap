// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchoolFilter renders correctly with props and store 1`] = `
<transition-stub name="modal">
  <div class="modal-mask">
    <div class="modal-wrapper">
      <div class="modal-container">
        <div class="modal-header">
          <h2>School filter</h2>
        </div>
        <div class="modal-body">
          <form>
            <div class="pl-row">
              <div class="pl-col">
                <label for="s_edu_sector">Education sector</label>
                <select id="s_edu_sector">
                  <option value="all">All</option>
                  <option value="Government">Government</option>
                  <option value="NonGovernment">Non Government</option>
                </select>
              </div>
            </div>
            <div class="pl-row">
              <div class="pl-col">
                <label class="checkbox" for="s_show_primary_schools">
                  <input id="s_show_primary_schools" type="checkbox"> Show primary schools
                </label>
              </div>
            </div>
            <div class="pl-row">
              <div class="pl-col">
                <label for="s_prim_rating">Overall score</label>
                <select id="s_prim_rating">
                  <option value="100">= 100</option>
                  <option value="99">&gt;= 99</option>
                  <option value="98">&gt;= 98</option>
                  <option value="97">&gt;= 97</option>
                  <option value="96">&gt;= 96</option>
                  <option value="95">&gt;= 95</option>
                  <option value="94">&gt;= 94</option>
                  <option value="93">&gt;= 93</option>
                  <option value="92">&gt;= 92</option>
                  <option value="91">&gt;= 91</option>
                  <option value="90">&gt;= 90</option>
                  <option value="0">All schools</option>
                </select>
              </div>
            </div>
            <div class="pl-row">
              <div class="pl-col">
                <label for="s_show_secondary_schools">
                  <input id="s_show_secondary_schools" type="checkbox"> Show secondary schools
                </label>
              </div>
            </div>
            <div class="pl-row">
              <div class="pl-col">
                <label for="s_sec_rating">Overall score</label>
                <select id="s_sec_rating">
                  <option value="100">= 100</option>
                  <option value="99">&gt;= 99</option>
                  <option value="98">&gt;= 98</option>
                  <option value="97">&gt;= 97</option>
                  <option value="96">&gt;= 96</option>
                  <option value="95">&gt;= 95</option>
                  <option value="94">&gt;= 94</option>
                  <option value="93">&gt;= 93</option>
                  <option value="92">&gt;= 92</option>
                  <option value="91">&gt;= 91</option>
                  <option value="90">&gt;= 90</option>
                  <option value="0">All schools</option>
                </select>
              </div>
            </div>
          </form>
        </div>
        <div class="modal-footer text-right">
          <button class="pl-button full small">
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</transition-stub>
`;
